backend/question_service/controllers/questionController.js

```diff
@@ -181,10 +182,55 @@ const deleteQuestion = async (req, res) => {
   }
 };
 
+// get questions from an array of question ids
+const getQuestions = async (req, res) => {
+  try {
+    const questionIds = req.body.questionIds;
+
+    if (!questionIds) {
+      return res.status(400).json({
+        message: "Question IDs is (are) required",
+      });
+    }
+
+    const questionDataArray = []; // push each qn data into here
+
+    for (const questionId of questionIds) {
+      const questionDoc = await db
+        .collection("questions")
+        .doc(questionId)
+        .get();
+
+      const questionData = questionDoc.data();
+
+      if (!questionData) {
+        console.log("Question doesn't exists!");
```

Comment:
![image](https://github.com/user-attachments/assets/0b8e047a-28b4-4b78-aa28-8ec7bb04738e)
thanks for pointing this out ! yes it would be more appropriate, now checks if response exists before accessing data